### PR TITLE
Fixes #14260: Ensure the leading dot is removed from dist in repos RPM

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: repos/katello-repos.spec [hunk 1]
```diff
@@ -46,7 +46,8 @@ install -m 644 %{SOURCE1} %{buildroot}%{_sysconfdir}/yum.repos.d/
 install -m 644 %{SOURCE2} %{buildroot}%{_sysconfdir}/pki/rpm-gpg/RPM-GPG-KEY-katello
 
 for repofile in %{buildroot}%{_sysconfdir}/yum.repos.d/*.repo; do
-    sed -i 's/@DIST@/%{dist}/' $repofile
+    trimmed_dist=`echo %{dist} | sed 's/.//'`
```
Comment (thomasmckay) [inline repos/katello-repos.spec]: This will trim the first character always. 's/.//' would just strip '.'. Not sure if that matters here or not.

**Comment by jlsherrill** [comment]
APJ

File: repos/katello-repos.spec (hunk 1 above)
Comment (thomasmckay) [inline repos/katello-repos.spec]: Sorry, should be 's/^.//' if just wish to trim leading '.'.

Comment (ehelms) [inline repos/katello-repos.spec]: I am confused -- we want to strip just the dot off the front.

Comment (thomasmckay) [inline repos/katello-repos.spec]:
```
[vagrant@katello foreman]$ echo "abc" | sed 's/.//'
bc
[vagrant@katello foreman]$ echo "abc" | sed 's/\.//'
abc
[vagrant@katello foreman]$ echo "a.bc" | sed 's/\.//'
abc
[vagrant@katello foreman]$ echo "a.bc" | sed 's/^\.//'
a.bc
[vagrant@katello foreman]$ echo ".a.bc" | sed 's/^\.//'
a.bc
```

I'm saying that this is the correct sed:

```
sed 's/^\.//'
```

Comment (thomasmckay) [inline repos/katello-repos.spec]: Sorry, I see my earlier comments lost a slash in the comment rendering, hence the confusion.

**Comment by ehelms** [comment]
Updated per @thomasmckay suggestion

**Comment by thomasmckay** [comment]
APJ

**Comment by thomasmckay** [comment]
APJ if don't care what first character being removed is

**Comment by ehelms** [comment]
I updated per IRC discussion to remove just the first explicit dot.

**Pull request merged**